### PR TITLE
Add analytics to permission system

### PR DESCRIPTION
**Pull request by jpcloureiro**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #mobile-planning/521

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/UI/NetworkList/index.js [hunk 1]
```diff
@@ -200,6 +200,7 @@ export class NetworkList extends PureComponent {
   onNetworkChange = (type) => {
     this.handleNetworkSelected(type, ETH, type);
     const { NetworkController, CurrencyRateController } = Engine.context;
+    const source = 'Wallet tab || Browser tab';
```
Comment (Cal-L) [inline app/components/UI/NetworkList/index.js:203]: Can probably pass the `source` property to `toggleNetworkModal` redux action, which the `NetworkList` component can consume

File: app/components/Views/AccountSelector/AccountSelector.tsx [hunk 2]
```diff
@@ -45,14 +45,29 @@ const AccountSelector = ({ route }: AccountSelectorProps) => {
...
+        return 'Undefined fromView';
+    }
+  };
+
   const _onSelectAccount = (address: string) => {
     const { PreferencesController } = Engine.context;
     PreferencesController.setSelectedAddress(address);
     sheetRef.current?.hide();
     onSelectAccount?.(address);
+    const source = getMainTab(route.params?.fromView);
```
Comment (Cal-L) [inline app/components/Views/AccountSelector/AccountSelector.tsx:66]: Can probably extract value to be hardcoded `Wallet tab` - Can reuse this extraction in other places to maintain consistency

File: app/components/Views/AccountSelector/AccountSelector.tsx [hunk 3]
```diff
@@ -45,14 +45,29 @@ const AccountSelector = ({ route }: AccountSelectorProps) => {
...
     InteractionManager.runAfterInteractions(() => {
       // Track Event: "Switched Account"
       AnalyticsV2.trackEvent(MetaMetricsEvents.SWITCHED_ACCOUNT, {
+        source,
```
Comment (Cal-L) [inline app/components/Views/AccountSelector/AccountSelector.tsx:70]: Also add `SWITCHED_ACCOUNT` to Browser tab/Permission UI

**Review by Cal-L — COMMENTED** [review]
Left some comments

File: app/components/Views/AccountSelector/AccountSelector.tsx (hunk 3 above)
Comment (jpcloureiro) [reply app/components/Views/AccountSelector/AccountSelector.tsx]: addressed in b2a7378f1cfca120c2e7756374fb9ede17d5815b

File: app/components/Views/AccountSelector/AccountSelector.tsx (hunk 2 above)
Comment (jpcloureiro) [reply app/components/Views/AccountSelector/AccountSelector.tsx]: addressed in b2a7378f1cfca120c2e7756374fb9ede17d5815b

File: app/components/UI/NetworkList/index.js (hunk 1 above)
Comment (jpcloureiro) [reply app/components/UI/NetworkList/index.js]: addressed in b2a7378f1cfca120c2e7756374fb9ede17d5815b

File: app/components/Views/Browser/index.js [hunk 4]
```diff
@@ -85,6 +87,8 @@ const Browser = (props) => {
   };
 
   const switchToTab = (tab) => {
+    //TODO: add here switch tab analytic event
```
[the remark this answers is not in the document]
Comment (jpcloureiro) [reply app/components/Views/Browser/index.js]: addressed in b2a7378f1cfca120c2e7756374fb9ede17d5815b

File: app/core/Analytics/MetaMetrics.events.ts [hunk 5]
```diff
@@ -246,6 +247,57 @@ enum EVENT_NAME {
   SCREENSHOT_WARNING = 'Screenshot Warning Displayed',
   SCREENSHOT_LEARN_MORE = 'Clicked Screenshot Warning Learn More',
   SCREENSHOT_OK = 'Clicked Screenshot Warning OK',
+
+  // Permission request
+  PERMISSION_REQUEST_STARTED = 'Permission Request Started',
+  PERMISSION_REQUEST_COMPLETED = 'Permission Request Completed',
+  PERMISSION_REQUEST_CANCELLED = 'Permission Request Cancelled',
```
[the remark this answers is not in the document]
Comment (jpcloureiro) [reply app/core/Analytics/MetaMetrics.events.ts]: addressed in b2a7378f1cfca120c2e7756374fb9ede17d5815b

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**